### PR TITLE
For #18240: Adjust preference screen to fit UX specs.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

Note: The "add credit card" and "manage saved cards" will not both be shown in the same time.
### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mcarare** [comment]
@gabrielluong I have removed the custom preference layout as we do not need it if we follow UX specs.

File: app/src/main/res/layout/preference_credit_cards_add_credit_card.xml [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent"
-        app:srcCompat="@drawable/ic_new" />
-
-    <TextView
-        android:id="@+id/add_credit_card_text"
-        android:layout_width="0dp"
-        android:layout_height="wrap_content"
-        android:layout_marginStart="@dimen/credit_cards_add_credit_card_text_horizontal_margin"
```
Comment (gabrielluong) [inline app/src/main/res/layout/preference_credit_cards_add_credit_card.xml:31]: Can we also remove this unused variable as well? 

`credit_cards_add_credit_card_text_horizontal_margin`,
`credit_cards_add_credit_card_text_size`

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -219,8 +219,6 @@
     <dimen name="credit_cards_add_credit_card_button_height">48dp</dimen>
     <!-- The horizontal margin for the "Add credit card" text.
          This should be the same value as radio_button_padding_horizontal. -->
```
Comment (gabrielluong) [inline app/src/main/res/values/dimens.xml:221]: Sorry, missed calling out `credit_cards_add_credit_card_button_height` is not needed as well I think. We can also remove their associated comments.

**Review by gabrielluong — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -214,13 +214,8 @@
     <dimen name="top_sites_text_margin_top">8dp</dimen>
 
     <!-- Credit Cards Fragment -->
-    <!-- The height of the "Add credit card" button in the Credit Cards fragment.
-         This should be the same value as search_engine_radio_button_height. -->
-    <dimen name="credit_cards_add_credit_card_button_height">48dp</dimen>
     <!-- The horizontal margin for the "Add credit card" text.
          This should be the same value as radio_button_padding_horizontal. -->
```
Comment (gabrielluong) [inline app/src/main/res/values/dimens.xml:218]: Sorry, this should also be removed.

```suggestion
```

**Review by gabrielluong — APPROVED** [review]

File: app/src/main/res/values/dimens.xml (hunk 3 above)
Comment (mcarare) [reply app/src/main/res/values/dimens.xml]: No problem! I looked into the history and did not know exactly what it was referring to. Thanks for clearing things up!

**Pull request merged**